druid-shell/src/mac/mod.rs

```diff
@@ -517,11 +521,31 @@ impl IdleHandle {
     }
 }
 
-fn get_characters(event: id) -> String {
+fn make_key_event(event: id) -> KeyData {
     unsafe {
-        let characters = event.characters();
-        let slice =
-            std::slice::from_raw_parts(characters.UTF8String() as *const _, characters.len());
-        std::str::from_utf8_unchecked(slice).to_owned()
+
+        let chars = event.characters();
+        let slice = std::slice::from_raw_parts(chars.UTF8String() as *const _, chars.len());
+        let text = std::str::from_utf8_unchecked(slice);
+
+        let unmodified_chars = event.charactersIgnoringModifiers();
+        let slice = std::slice::from_raw_parts(unmodified_chars.UTF8String() as *const _, unmodified_chars.len());
+        let unmodified_text = std::str::from_utf8_unchecked(slice);
+
+        let virtual_key: std::os::raw::c_ushort = msg_send!(event, keyCode);
+
+        let is_repeat: bool = msg_send!(event, isARepeat);
```

Comment:
this is done, but it will be a while before we get a release and I don't think it's worth tracking the crate on github?

https://github.com/servo/core-foundation-rs/commit/0d56822550edfb13ff213b8454aa9cf59d6d592e